src/auxiliary/auxiliary.jl

```diff
@@ -345,4 +345,22 @@ function register_error_hints()
 
     return nothing
 end
+
+"""
+    retrieve(file_path, src_url)
+
+Download file from given url to given path.
+Avoids race condition when multiple MPI ranks are used.
+"""
+function retrieve(file_path, src_url)
+    if mpi_isroot()
+        isfile(file_path) || download(src_url, file_path)
+    end
+
+    if mpi_isparallel()
+        MPI.Barrier(mpi_comm())
+    end
+
+    return file_path
+end
```

Comment:
I think it's kind of confusing that this is very similar to `download` - but the other way round. Could we maybe change this to something like

```suggestion
"""
    Trixi.download(src_url, file_path)

Download a file from given `src_url` to given `file_path` if
`file_path` is not already a file.
This is a small wrapper of `Downloads.download(src_url, file_path)`
that avoids race conditions when multiple MPI ranks are used.
"""
function download(src_url, file_path)
    # Note that `mpi_isroot()` is also `true` if running
    # in serial (without MPI).
    if mpi_isroot()
        isfile(file_path) || Downloads.download(src_url, file_path)
    end

    if mpi_isparallel()
        MPI.Barrier(mpi_comm())
    end

    return file_path
end
```